integration_test/pathfilter_integration_test.go

```diff
@@ -0,0 +1,112 @@
+package integrationtest
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/node-disk-manager/integration_test/ndm_util"
+	"time"
+)
+
+var _ = Describe("Path filter integration Test", func() {
+
+	var configMapPatch ndmutil.ConfigMapPatch
+	var err error
+
+	When("Path filter is disabled", func() {
+		err = ndmutil.InitEnvironment()
+		Expect(err).NotTo(HaveOccurred())
+
+		configMapPatch = ndmutil.GetNDMConfig(ndmutil.GetNDMOperatorFilePath())
+		configMapPatch.SetPathFilter("false")
+
+		ndmutil.ReplaceAndApplyConfig(configMapPatch)
+		// It waits till node-disk-manager is ready or timeout reached
+		err = ndmutil.WaitTillNDMisUpOrTimeout(5 * time.Minute)
+		Expect(err).NotTo(HaveOccurred())
+
+		It("has matching disks inside node and the host", func() {
+
+			matched, err := ndmutil.MatchDisksOutsideAndInside()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(matched).To(BeTrue())
+		})
+	})
+
+	When("A device is included", func() {
```

Comment:
From the code I could see that we are testing at least two groups of tests, one related to device and one test for disabled Path Filter, so just check if it would be better to separate the groups out.

Reply:
Ohkay got it.

Reply:
I saw it. Its real good.